packages/utils/spatial-utils/src/spatial.js

```diff
@@ -316,3 +316,39 @@ export async function initializeRasterLayersAndChannels(
   const autoImageLayerDefs = await Promise.all(autoImageLayerDefPromises);
   return [autoImageLayerDefs, nextImageLoaders, nextImageMetaAndLayers];
 }
+
+/**
+ * Convert an array of coordinateTransformations objects to a 16-element
+ * plain JS array using Matrix4 linear algebra transformation functions.
+ * @param {object[]} coordinateTransformations List of objects matching the
+ * OME-NGFF v0.4 coordinateTransformations spec.
+ * @returns {number[]} Plain JS array of 16 numbers representing the Matrix4.
+ */
+export function coordinateTransformationsToMatrix(coordinateTransformations) {
+  let mat = (new Matrix4()).identity();
+  if (coordinateTransformations) {
+    // Applied sequentially and in order according to the OME-NGFF v0.4 spec.
+    // Reference: https://ngff.openmicroscopy.org/0.4/#trafo-md
+    coordinateTransformations.forEach((transform) => {
```

Comment:
Don't we need handling ffor the per-dimension aspect of things from https://ngff.openmicroscopy.org/0.4/#trafo-md i.e "The length of vector defines number of dimensions" so this vector could be 5 but we only want the xyz transformation (presumably).

Reply:
Good point - I will add code to account for the OME-NGFF v0.4 `axes` metadata. It is less clear what to do for the OME-TIFF case where I was allowing for the `coordinateTransformations` to be provided as part of the `options` in the Vitessce config - doing an analogous check of the dimension ordering would require loading the OME-XML metadata. I think there are two ways forward for that case:
- we can either assume XYZ when using `coordinateTransformations` in the OME-TIFF case, or
- we can load the OME-XML and implement the same dimension-order matching procedure to match the OME-NGFF behavior

Reply:
Good point - I have updated to use `DimensionOrder` from the OME-XML